packages/@uppy/companion/src/server/provider/box/index.js

```diff
@@ -93,15 +92,15 @@ class Box extends Provider {
 
   async size ({ id, token }) {
     return this.#withErrorHandling('provider.box.size.error', async () => {
-      const { size } = await getClient({ token }).get(`files/${id}`, { responseType: 'json' }).json()
+      const { size } = await (await getClient({ token })).get(`files/${id}`, { responseType: 'json' }).json()
```

Comment:
these double `await`'s aren't great and without TS you can easily forgot them and introduce a bug. Not sure what to do about it though. 

Can't we put the client inside middleware context, making it available everywhere?

Reply:
I don't think middleware context is more robust now that we don't have typescript (middlewares are harder to refactor than imports IMO). I don't see a big problem with this, as once we refactor to TS it'll be safe. and even before refactoring, it will crash during runtime if one forgets `await`. BTW once we refactor to ESM, we won't need these `await` at all.

Reply:
Since it's used in all providers, I think middleware would work well. With unplanned statements like "once we refactor to TS" and "once we refactor to ESM" I usually get a hunch that it's something that stays forever. Ideally we write code for what works well now, not what may or may not happen in the future.

Reply:
Here's what happens if I try to remove the `await`:

```console
$ yarn build:companion
src/server/provider/box/index.js:63:45 - error TS2339: Property 'stream' does not exist on type 'Promise<Got>'.

63       const stream = (getClient({ token })).stream.get(`files/${id}/content`, { responseType: 'json' })
                                               ~~~~~~

  src/server/provider/box/index.js:63:45
    63       const stream = (getClient({ token })).stream.get(`files/${id}/content`, { responseType: 'json' })
                                                   ~~~~~~
    Did you forget to use 'await'?
```

I think TS got out back.